### PR TITLE
Create SineGenerator.java and MixedGenerator.java

### PR DESCRIPTION
**Pull request by pmgj**
Two stream generators as proposed by "Gama, Joao, et al. "Learning with drift detection." Advances in artificial intelligence–SBIA 2004. Springer Berlin Heidelberg, 2004. 286-295."

### CODE REVIEW

File: moa/src/main/java/moa/streams/generators/MixedGenerator.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+import moa.options.AbstractOptionHandler;
+import moa.streams.InstanceStream;
+import moa.tasks.TaskMonitor;
+
+/**
+ * Creates a stream consisting of two boolean and two numeric attributes and
+ * represents a drift by reversing the drift function. Based on proposal by
+ * "Gama, Joao, et al. "Learning with drift detection." Advances in artificial 
+ * intelligence–SBIA 2004. Springer Berlin Heidelberg, 2004. 286-295."
+ *
```
Comment (abifet) [inline moa/src/main/java/moa/streams/generators/MixedGenerator.java]: We may add the original explanation to make the code easier to understand:

MIXED. Abrupt concept drift, boolean noise-free examples. Four relevant
attributes, two boolean attributes v,w and two numeric attributes
from [0; 1]. The examples are classified positive if two of three conditions
are satisfied: v,w, y < 0,5 + 0,3  sin(3 PI x). After each context change the
classification is reversed.

File: moa/src/main/java/moa/streams/generators/SineGenerator.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+import moa.options.AbstractOptionHandler;
+import moa.streams.InstanceStream;
+import moa.tasks.TaskMonitor;
+
+/**
+ * Stream generator for the problem of identifying a Sine curve. Based on
+ * Sine1, Sine2, Sinirrel1, and Sinirrel2 as proposed by
+ * "Gama, Joao, et al. "Learning with drift detection." Advances in artificial 
+ * intelligence–SBIA 2004. Springer Berlin Heidelberg, 2004. 286-295."
+ *
```
Comment (abifet) [inline moa/src/main/java/moa/streams/generators/SineGenerator.java]: We may also add the original explanation to make the code easier to understand:

SINE1. Abrupt concept drift, noise-free examples. It has
two relevant attributes. Each attributes has values uniformly distributed in
[0; 1]. In the first context all points below the curve y = sin(x) are classified
as positive. After the context change the classification is reversed.
2. SINE2. The same two relevant attributes. The classification function is y <
0,5 + 0,3 sin(3 PI x). After the context change the classification is reversed.
3. SINIRREL1. Presence of irrelevant attributes. The same classification
function of SINE1 but the examples have two more random attributes
with no influence on the classification function.
4. SINIRREL2. The same classification function of SINE2 but the examples
have two more random attributes with no influence on the classification
function.

**Comment by pmgj** [comment]
The description of the generators have been updated to the original explanation.

**Comment by abifet** [comment]
Are the new descriptions interchanged between the two files?

**Comment by pmgj** [comment]
Sorry for the mistake. The descriptions have been corrected.

**Comment by abifet** [comment]
+1

**Pull request merged**